static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx

```diff
@@ -5,6 +5,45 @@ import {useEffectAfterFirstRender} from 'sentry/utils/useEffectAfterFirstRender'
 import {VirtualizedTree} from './VirtualizedTree';
 import {VirtualizedTreeNode} from './VirtualizedTreeNode';
 
+type AnimationTimeoutId = {
+  id: number;
+};
+
+const cancelAnimationTimeout = (frame: AnimationTimeoutId) =>
+  window.cancelAnimationFrame(frame.id);
+
+/**
+ * Recursively calls requestAnimationFrame until a specified delay has been met or exceeded.
+ * When the delay time has been reached the function you're timing out will be called.
+ * This was copied from react-virtualized, with credits to the original author.
+ *
+ * Credit: Joe Lambert (https://gist.github.com/joelambert/1002116#file-requesttimeout-js)
+ */
+const requestAnimationTimeout = (
+  callback: Function,
+  delay: number
+): AnimationTimeoutId => {
+  let start;
+  // wait for end of processing current event handler, because event handler may be long
+  Promise.resolve().then(() => {
+    start = Date.now();
+  });
+
+  const timeout = () => {
+    if (Date.now() - start >= delay) {
```

Comment:
could `start` be `undefined` here if this handler is called before the `.then` is called?